pandas/indexes/multi.py

```diff
@@ -1007,6 +1008,9 @@ def from_tuples(cls, tuples, sortorder=None, names=None):
             # I think this is right? Not quite sure...
             raise TypeError('Cannot infer number of levels from empty list')
 
+        if not _check_equal_length(tuples):
```

Comment:
I'm fine with that.  Would it make sense to move `_check_equal_length` to lib.pyx?